# 02 Trend Across Time

## import modules and dataset

In [11]:
import pandas as pd
import altair as alt
# alt.renderers.enable('jupyterlab')

In [12]:
trials_clean = pd.read_csv('./export/trials_clean.csv')
trials_clean.sample(20)

year  decade  century  tried  deaths             city  \
3890   1630.0    1630     1600      1     0.0              NaN   
6414   1384.0    1380     1300      1     0.0            Milan   
1150   1629.0    1620     1600      1     NaN     Breitenbrunn   
7882   1598.0    1590     1500      1     1.0           Cassel   
5649   1677.0    1670     1600      1     0.0              NaN   
10871  1639.0    1630     1600      1     0.0  Midden-Delfland   
3265   1614.0    1610     1600      1     1.0              NaN   
2329   1626.0    1620     1600      1     0.0             Ohey   
6302   1366.0    1360     1300      1     1.0              NaN   
4719   1659.0    1650     1600      1     1.0              NaN   
8824   1642.0    1640     1600      1     0.0         Valangin   
2202   1599.0    1590     1500      1     1.0          Eghezee   
5926   1702.0    1700     1700      1     NaN              NaN   
2838      NaN    1610     1600      4     0.0         Bouillon   
10370  1645.0    1640     1600      1     NaN              NaN   
6547   1336.0    1330     1300      1     NaN          Avignon   
5151   1662.0    1660     1600      1     0.0              NaN   
277       NaN    1550     1500      1     0.0              NaN   
7719   1679.0    1670     1600      2     2.0         Freiburg   
8768   1623.0    1620     1600      1     0.0        Colombier   

                    gadm.adm2                   gadm.adm1       gadm.adm0  \
3890                      NaN                    Scotland  United Kingdom   
6414                   Milano                   Lombardia           Italy   
1150            Mittelfranken                      Bayern         Germany   
7882                     Nord          Nord-Pas-de-Calais          France   
5649      West Dunbartonshire                    Scotland  United Kingdom   
10871             Schipluiden                Zuid-Holland     Netherlands   
3265   Perthshire and Kinross                    Scotland  United Kingdom   
2329                    Namur                    Wallonie         Belgium   
6302                      NaN                     England  United Kingdom   
4719           South Ayrshire                    Scotland  United Kingdom   
8824               Val-de-Ruz                   Neuchatel     Switzerland   
2202                    Namur                    Wallonie         Belgium   
5926    Dumfries and Galloway                    Scotland  United Kingdom   
2838               Luxembourg                    Wallonie         Belgium   
10370                   Essex                     England  United Kingdom   
6547                 Vaucluse  Provence-Alpes-Cote d'Azur          France   
5151             Renfrewshire                    Scotland  United Kingdom   
277                  Willisau                     Lucerne     Switzerland   
7719                 Freiburg           Baden-Wurttemberg         Germany   
8768                   Boudry                   Neuchatel     Switzerland   

             lon        lat          record.source  
3890         NaN        NaN  Goodare et al. (2003)  
6414    9.185924  45.465422      Kieckhefer (1976)  
1150   11.622700  49.085300       Behringer (1987)  
7882         NaN        NaN       Monballyu (2002)  
5649         NaN        NaN  Goodare et al. (2003)  
10871        NaN        NaN          Waardt (1991)  
3265         NaN        NaN  Goodare et al. (2003)  
2329    5.124600  50.436700        Brouette (1953)  
6302         NaN        NaN      Kieckhefer (1976)  
4719         NaN        NaN  Goodare et al. (2003)  
8824    6.906460  47.016200          Monter (1976)  
2202    4.903650  50.592600        Brouette (1953)  
5926         NaN        NaN  Goodare et al. (2003)  
2838    5.070760  49.794700  Dupont-Bouchat (1978)  
10370        NaN        NaN        Valletta (2000)  
6547    4.805528  43.949317      Kieckhefer (1976)  
5151         NaN        NaN  Goodare et al. (2003)  
277          NaN        NaN           Bader (1945)

### First glance: total casualty
In this dataset:

In [13]:
# how many people are tried in total
# df.sum(axis=0) Summing columns, or summing across rows, set axis=0
trials_clean['tried'].sum(axis=0)

43240

In [14]:
# how many people are killed in total
trials_clean['deaths'].sum(axis=0)

16333.0

So this is just a subset of the people tried and persecuted during the witch trial. According to [Wikipedia's Witch Trials in the early modern period](https://en.wikipedia.org/wiki/Witch_trials_in_the_early_modern_period), between 1400 to 1782, around 40,000 to 60,000 people were killed due to suspicion that they were practicing witchcraft. Some sources estimate that a total of 100,000 trials occurred at its maximum for a similar period.

## Q1: What time period (century/decade/year) witnessed the peak of witch trials?

### Witch Trials by century - line chart

In [15]:
# Group by century 
trials_by_century = trials_clean.groupby([pd.Grouper(key='century')]).count()[['tried']].reset_index()

# rename the columns
trials_by_century.columns = ['century', 'trials_count']
trials_by_century

century  trials_count
0     1300           150
1     1400           607
2     1500          2364
3     1600          7164
4     1700           652
5     1800             3

In [16]:
# line chart by altair
alt.Chart(trials_by_century).mark_line().encode(
    x=alt.X('century'),
    y=alt.Y('trials_count'),
).properties(
    title={
        'text': 'Witch Trials by century',
    }
)

/Users/direwolfsummer/.pyenv/versions/3.9.4/envs/j233-files-3.9.4/lib/python3.9/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


<VegaLite 4 object>

If you see this message, it means the renderer has not been properly enabled
for the frontend that you are using. For more information, see
https://altair-viz.github.io/user_guide/troubleshooting.html


#### The zenith between 1560 and 1630
The period between 1560 and 1630 was known as the 'Great Hunt', when 60% of the prosecution took place. Over the course of a century and a half, 80,000 people were tried for witchcraft and half of them were executed, often burned alive. This time period coincided with the Counter-Reformation and the European wars of religion. [Economists Peter T. Leeson and Jake Russ argued](https://www.peterleeson.com/witch_trials.pdf) that the surgence of European witch trials reflected non-price competition between the Catholic and Protestant churches for religious market share in confessionally contested parts of Christendom.

Simply put, as competing Catholic and Protestant churches vied to win over or retain their followers, they needed to make an impact – and witch trials were the battleground they chose ([the Guardian](https://www.theguardian.com/society/2018/jan/07/witchcraft-economics-reformation-catholic-protestant-market-share)).

### Witch Trials By Decade - line chart

In [17]:
# Group by decade 
trials_by_decade = trials_clean.groupby([pd.Grouper(key='decade')]).count()[['tried']].reset_index()

# rename the columns
trials_by_decade.columns = ['decade', 'trials_count']
trials_by_decade

decade  trials_count
0     1300            11
1     1310            21
2     1320            21
3     1330            25
4     1340             8
5     1350             4
6     1360             3
7     1370            15
8     1380            17
9     1390            25
10    1400            32
11    1410            20
12    1420            35
13    1430            52
14    1440            57
15    1450            90
16    1460            65
17    1470            68
18    1480           101
19    1490            87
20    1500            39
21    1510            43
22    1520            64
23    1530            73
24    1540           106
25    1550           111
26    1560           309
27    1570           291
28    1580           373
29    1590           955
30    1600           621
31    1610           878
32    1620           934
33    1630           724
34    1640          1111
35    1650           872
36    1660          1050
37    1670           419
38    1680           260
39    1690           295
40    1700           195
41    1710           107
42    1720           105
43    1730            68
44    1740            89
45    1750            62
46    1760             8
47    1770            11
48    1780             3
49    1790             4
50    1820             1
51    1830             1
52    1850             1

In [18]:
# line chart by altair
alt.Chart(trials_by_decade).mark_line().encode(
    x=alt.X('decade'),
    y=alt.Y('trials_count'),
).properties(
    title={
        'text': 'Witch Trials by decade',
    }
)

/Users/direwolfsummer/.pyenv/versions/3.9.4/envs/j233-files-3.9.4/lib/python3.9/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


<VegaLite 4 object>

If you see this message, it means the renderer has not been properly enabled
for the frontend that you are using. For more information, see
https://altair-viz.github.io/user_guide/troubleshooting.html


### Witch Trials By Year - line chart

In [19]:
# Group by year 
trials_by_year = trials_clean.groupby([pd.Grouper(key='year')]).count()[['tried']].reset_index()

# rename the columns
trials_by_year.columns = ['year', 'trials_count']

In [20]:
# line chart by altair
alt.Chart(trials_by_year).mark_line().encode(
    x=alt.X('year'),
    y=alt.Y('trials_count'),
).properties(
    title={
        'text': 'Witch Trials by year',
    }
)

/Users/direwolfsummer/.pyenv/versions/3.9.4/envs/j233-files-3.9.4/lib/python3.9/site-packages/altair/utils/core.py:317: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for col_name, dtype in df.dtypes.iteritems():


<VegaLite 4 object>

If you see this message, it means the renderer has not been properly enabled
for the frontend that you are using. For more information, see
https://altair-viz.github.io/user_guide/troubleshooting.html


According to the data we have, the 16 century witnessed the highest witch trials count. There are 1050 cases in 1660 alone. 